code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,448 @@
+	var/power_format_multi = 1
+	/// Same as above but for output
+	var/power_format_multi_output = 1
+
+	/// Are we selling the energy or just sending it into the ether
+	var/selling_energy = FALSE
+
+	/// How much energy have we sold in total (Joules)
+	var/total_energy = 0
+	/// How much energy do you have to sell in order to get an announcement
+	var/announcement_threshold = 1 MJ
+
+	/// How much credits we have earned in total
+	var/total_earnings = 0
+	/// The amount of money we haven't sent to cargo yet
+	var/unsent_earnings = 0
+
+	/// How much we are inputing pre multiplier
+	var/input_number = 0
+	/// How much we are outputting pre multiplier
+	var/output_number = 0
+	/// Our set input pulling
+	var/input_pulling = 0
+	/// Announcement configuration for updates
+	var/datum/announcer/announcer = new(config_type = /datum/announcement_configuration/ptl)
+	/// Last direction the laser was pointing. So offset doesn't get handles when it doesn't need to
+	var/last_dir = 0
+
+
+/obj/machinery/power/transmission_laser/Initialize(mapload)
+	. = ..()
+	range = get_dist(get_step(get_front_turf(), dir), get_edge_target_turf(get_front_turf(), dir))
+	if(!powernet)
+		connect_to_network()
+	handle_offset()
+	update_icon()
+
+/obj/machinery/power/transmission_laser/proc/handle_offset()
+	switch(dir)
+		if(NORTH)
+			pixel_x = -64
+			pixel_y = 0
+			var/datum/component/oldmap = GetComponent(/datum/component/multitile)
+			if(oldmap)
+				qdel(oldmap)
+			AddComponent(/datum/component/multitile, list(
+				list(0, 1, 			0,		),
+				list(1, 1, 			1,		),
+				list(1, 1, 		MACH_CENTER),
+			))
+		if(SOUTH)
+			pixel_x = 0
+			pixel_y = -64
+			var/datum/component/oldmap = GetComponent(/datum/component/multitile)
+			if(oldmap)
+				qdel(oldmap)
+			AddComponent(/datum/component/multitile, list(
+				list(MACH_CENTER, 	1, 1),
+				list(1, 			1, 1),
+				list(0, 			1, 0),
+			))
+		if(WEST)
+			pixel_x = -64
+			pixel_y = 0
+			var/datum/component/oldmap = GetComponent(/datum/component/multitile)
+			if(oldmap)
+				qdel(oldmap)
+			AddComponent(/datum/component/multitile, list(
+				list(0, 1, 			1,		),
+				list(1, 1, 			1,		),
+				list(0, 1, 		MACH_CENTER),
+			))
+		if(EAST)
+			pixel_x = 0
+			pixel_y = 0
+			var/datum/component/oldmap = GetComponent(/datum/component/multitile)
+			if(oldmap)
+				qdel(oldmap)
+			AddComponent(/datum/component/multitile, list(
+				list(1,				1, 0),
+				list(1,				1, 1),
+				list(MACH_CENTER, 	1, 0),
+			))
+
+/obj/machinery/power/transmission_laser/Destroy()
+	. = ..()
+	if(length(laser_effects))
+		destroy_lasers()
+	blocked_objects = null
+
+/obj/machinery/power/transmission_laser/proc/get_back_turf()
+	//this is weird as i believe byond sets the bottom left corner as the source corner like
+	// x-x-x
+	// x-x-x
+	// o-x-x
+	//which would mean finding the true back turf would require centering than taking a step in the inverse direction
+	var/turf/center = locate(x + 1, y + 1, z)
+	if(!center)///what
```

Comment:
?
```suggestion
	if(!center) //what
```

Reply:
Honestly this check is quite redundant. If locate is failing here we have bigger issues, and I don't think the front turf being improperly set is handled well anyhow.

Reply:
I'll just remove it.